src/modules/ce/components/ClientCeEligibility.tsx

```diff
@@ -0,0 +1,134 @@
+import { emptyErrorState, ErrorState, hasErrors } from '@/modules/errors/util';
+import { FormDefinitionHandlers } from '@/modules/form/hooks/useFormDefinitionHandlers';
+import { DynamicInputCommonProps } from '@/modules/form/types';
+import { localResolvePickList } from '@/modules/form/util/formUtil';
+import {
+  CalculateClientCeEligibilityMutation,
+  PickListOption,
+  useCalculateClientCeEligibilityMutation,
+} from '@/types/gqlTypes';
+
+const projectTypePickList = localResolvePickList('ProjectType');
+
+interface ClientEligibilityProps extends DynamicInputCommonProps {
+  value?: PickListOption[] | null;
+  onChange?: (value: PickListOption[] | null) => void;
+  handlers?: FormDefinitionHandlers;
+}
+
+// Shows a button to calculate client eligibility based on current form values, and displays the result.
+const ClientCeEligibility = ({
+  value,
+  onChange,
+  label,
+  disabled = false,
+  handlers,
+}: ClientEligibilityProps) => {
+  const [errorState, setErrorState] = useState<ErrorState>(emptyErrorState);
+
+  const [calculateClientEligibility, { loading }] =
+    useCalculateClientCeEligibilityMutation({
+      onCompleted: (data: CalculateClientCeEligibilityMutation) => {
+        const errors = data.calculateClientCeEligibility?.errors || [];
+        if (errors.length > 0) {
+          setErrorState({ ...emptyErrorState, errors });
+          return;
+        }
+
+        setErrorState(emptyErrorState);
+
+        if (onChange && data.calculateClientCeEligibility?.projectTypes) {
+          const eligibleProjectTypes =
+            data.calculateClientCeEligibility.projectTypes;
+
+          // Convert the returned project type codes to their corresponding pick list options.
+          // Filter out undefined values to keep typescript happy
+          const matchingPickListOptions = eligibleProjectTypes
+            .map((projectTypeCode) =>
+              projectTypePickList?.find(
+                (pickListOption) => pickListOption.code === projectTypeCode
+              )
+            )
+            .filter((option): option is PickListOption => option !== undefined);
+
+          onChange(matchingPickListOptions);
+        }
+      },
+      onError: (apolloError: any) => {
+        setErrorState({ ...emptyErrorState, apolloError });
+      },
+    });
+
+  const { formDefinitionIdentifier, enrollmentId } =
+    useContext(AssessmentContext) || {};
+
+  const hasRequiredContext = !!enrollmentId && !!formDefinitionIdentifier;
+
+  const handleFetch = useCallback(() => {
+    if (hasRequiredContext) {
+      calculateClientEligibility({
+        variables: {
+          enrollmentId,
+          formDefinitionIdentifier: formDefinitionIdentifier,
+          valuesByLinkId: handlers?.getValuesForSubmit().valuesByLinkId || {},
+        },
+      });
+    }
+  }, [
+    hasRequiredContext,
+    calculateClientEligibility,
+    enrollmentId,
+    formDefinitionIdentifier,
+    handlers,
```

Comment:
I'm not sure about depending on `handlers` here because I'm uncertain how often that changes. It might be safer to just depend on the specific function we're using:
```jsx
  const { getValuesForSubmit } = handlers; // assuming handlers is required
  const handleFetch = useCallback(() => {
    if (hasRequiredContext) {
      calculateClientEligibility({
        variables: {
          enrollmentId,
          formDefinitionIdentifier: formDefinitionIdentifier,
          valuesByLinkId: getValuesForSubmit().valuesByLinkId,
        },
      });
    }
  }, [
    hasRequiredContext,
    calculateClientEligibility,
    enrollmentId,
    formDefinitionIdentifier,
    getValuesForSubmit,
  ]);
```